mindsdb/integrations/handlers/sentence_transformer_handler/sentence_transformer_handler.py

```diff
@@ -0,0 +1,71 @@
+from typing import Optional
+
+import dill
+import pandas as pd
+from integrations.handlers.sentence_transformer_handler.settings import (
+    df_to_documents,
+    load_embeddings_model,
+)
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.utilities import log
+
+logger = log.get_log(__name__)
+
+
+class SentenceTransformerHandler(BaseMLEngine):
+    name = "sentence transformer"
+
+    def create(self, target, df=None, args=None, **kwargs):
+
+        args = args["using"]
+
+        model = load_embeddings_model(args["embeddings_model_name"])
+
+        self.model_storage.file_set("model", dill.dumps(model))
+        self.model_storage.json_set("args", args)
+
+    def predict(self, df, args=None):
+
+        args = args["predict_params"]
+        columns = args.get("columns")
+
+        if columns:
+            if isinstance(args["columns"], str):
+                columns = [columns]
+
+        else:
+            logger.info("no columns specified, all columns from input will be embedded")
+
+            columns = df.columns
+
+        documents = df_to_documents(df=df, page_content_columns=columns)
+
+        model = dill.loads(self.model_storage.file_get("model"))
+
+        embeddings = []
+        metadata = []
+
+        for _, document in enumerate(documents):
```

Comment:
Had a meeting recently with Yoshi and Tom, we've agreed that the output of an embedding model such as the ones created by this handler should always output a single embedding, i.e. a dataframe with a single column.

The aggregation mechanism is up to the underlying integration, as long as it is consistent it will be fine.

Reply:
Ok good to know, apologies I wasn't aware of this

Reply:
No worries, we actually dropped the ball on communicating this properly I'd say.

re: metadata yes, I think so!